x-pack/legacy/plugins/task_manager/task_runner.ts

```diff
@@ -211,27 +222,35 @@ export class TaskManagerRunner implements TaskRunner {
 
   private async processResultForRecurringTask(result: RunResult): Promise<RunResult> {
     // recurring task: update the task instance
+    const startedAt = this.instance.startedAt!;
     const state = result.state || this.instance.state || {};
-    const status = this.instance.attempts < this.store.maxAttempts ? 'idle' : 'failed';
+    const status = this.getInstanceStatus();
 
     let runAt;
     if (status === 'failed') {
       // task run errored, keep the same runAt
       runAt = this.instance.runAt;
+    } else if (result.runAt) {
+      runAt = result.runAt;
+    } else if (result.error) {
+      // when result.error is truthy, then we're retrying because it failed
+      runAt = this.definition.getRetryDelay
+        ? new Date(
+            Date.now() + this.definition.getRetryDelay(this.instance.attempts, result.error) * 1000
+          )
+        : minutesFromNow(this.instance.attempts * 5); // incrementally backs off an extra 5m per failure
```

Comment:
I wonder if we should just initialize `this.defininition.getRetryDelay()` to whatever would be appropriate, if null when passed in, to avoid some of these unwieldy `? :` operators

Reply:
Agreed, I think that makes better sense, I will implement the change.